index.c

```diff
@@ -1774,24 +1789,23 @@ int mutt_index_menu(void)
         break;
 
       case OP_MAIN_TAG_PATTERN:
-
-        CHECK_MSGCOUNT;
-        CHECK_VISIBLE;
+        if (!prereq(Context, menu, CHECK_IN_MAILBOX | CHECK_MSGCOUNT | CHECK_VISIBLE))
+          break;
         mutt_pattern_func(MUTT_TAG, _("Tag messages matching: "));
         menu->redraw |= REDRAW_INDEX | REDRAW_STATUS;
         break;
 
       case OP_MAIN_UNDELETE_PATTERN:
-
-        CHECK_MSGCOUNT;
-        CHECK_VISIBLE;
-        CHECK_READONLY;
-        /* L10N: CHECK_ACL */
-        /* L10N: Due to the implementation details we do not know whether we
-           undelete zero, 1, 12, ... messages. So in English we use
-           "messages". Your language might have other means to express this.
-         */
-        CHECK_ACL(MUTT_ACL_DELETE, _("Cannot undelete messages"));
+        if (!prereq(Context, menu, CHECK_IN_MAILBOX | CHECK_MSGCOUNT | CHECK_VISIBLE | CHECK_READONLY) ||
+            /* L10N: CHECK_ACL */
```

Comment:
I think we should move the comment above the conditional, it makes it easier to see the entire conditional.